dashboard-frontend/src/types/study.ts

```diff
@@ -0,0 +1,33 @@
+export interface Study {
+  id: string;
+  name: string;
+  slug: string;
+  createdAt: string;
+  progress: StudyProgress;
+  strata: Strata[];
+}
+
+interface StudyProgress {
+  currentParticipants: number;
+  expectedParticipants: number;
+  currentAverageDeviation: number;
+  expectedAverageDeviation: number;
+  date: string;
+  averageDeviationOverTime: DataPoint[];
+}
+
+interface Strata {
+  id: string;
+  percentageDesired: number;
+  currentPercentage: number;
+  remainingPercentage: number;
+  expectedPercentage: number;
+  currentPriceToAdquireParticipant: number;
+  currentBudgetDecided: number;
+  remainingPercentageOverTime: DataPoint[];
```

Comment:
I like the idea of having this, but I don't like nesting this. 

`StratumProgress` should have a datetime. 

Then you can have an array of `StratumProgress` which has _all_ values over time (you can select `remainingPercentage` easily if wanted to plot, or anything else!

Reply:
Ah, the same should apply for `StudyProgress` actually!

Reply:
(remove averageDeviationOverTime and instead anything should be able to be created "overTime" - it's just a view on any `StudyProgress[]`.

Reply:
Yes I will update the code 👍

Actually it crossed my mind to promote **StratumProgress** (before called Strata) into a data point.

But I was not sure about if we could aggregate all the data from an **StratumProgress** at an specific point in time.

I knew that we could for **remainingPercentage** because of this dashboard -> https://user-images.githubusercontent.com/7491490/119506873-9d6ab380-bd6e-11eb-8fbf-c2ff7caae8bc.png 
and that's why I decided to separe it.

About the **StudyProgress**, the same reasoning I know that we can aggregate the current average deviation at a very granular time intervals but was not sure about the other values.

Reply:
In the **StudyProgress** interface I also added an **id field**.

The frontend needs it in order to diff re-renders etc..

From the frontend side the **id** will be an opaque token (that is unique), implementation wise in the api we can reuse datetime or what we find more appropiate when we implement it.

Reply:
Yeah of course, sounds reasonable... Right now there is nothing enforced for uniqueness at DB level, but we can change that. A report should be unique. Maybe will add UUID at DB level... Will think about it...